src/hnc_worker_sup.erl

```diff
@@ -17,39 +17,33 @@
 
 -behavior(supervisor).
 
--export([start_link/3]).
--export([start_worker/1, stop_worker/2]).
+-export([start_link/4]).
 -export([init/1]).
+-export([start_worker/1, stop_worker/2]).
 
--spec start_link(module(), term(), hnc:shutdown()) -> {ok, pid()}.
-start_link(Mod, Args, Shutdown) ->
-	supervisor:start_link(?MODULE, {Mod, Args, Shutdown}).
+start_link(Mod, Args, Shutdown, Modules) ->
+	supervisor:start_link(?MODULE, {Mod, Args, Shutdown, Modules}).
 
--spec start_worker(pid()) -> {ok, hnc:worker()}.
 start_worker(Sup) ->
 	supervisor:start_child(Sup, []).
 
--spec stop_worker(pid(), hnc:worker()) -> ok.
 stop_worker(Sup, Worker) ->
-	supervisor:terminate_child(Sup, Worker).
+	supervisor:terminate_child(Sup, Worker).	
 
-init({Mod, Args, Shutdown}) ->
-	{module, Mod}=code:ensure_loaded(Mod),
-	Modules=case erlang:function_exported(Mod, get_modules, 0) of
-		true -> Mod:get_modules();
-		false -> [Mod]
-	end,
+init({Mod, Args, Shutdown, Modules}) ->
+	ok=logger:update_process_metadata(#{hnc => #{module => ?MODULE}}),
 	{
 		ok,
 		{
 			#{
-				strategy => simple_one_for_one
+				strategy => simple_one_for_one,
+				intensity => 0
 			},
 			[
 				#{
 					id => hnc_worker,
 					start => {Mod, start_link, [Args]},
-					restart => temporary,
+					restart => permanent,
```

Comment:
Why `permanent`, shouldn't it be `temporary`?

Reply:
This, and the logger fiddling, is because the supervisor should go away when the worker goes away. With temporary, the supervisor would not mind the worker crashing, with permanent it would attempt a restart. But, by setting intensity to zero, it can't and will itself crash. But it will generate a max_restart_intensity_reached error in the logs, which gets filtered out with the logger filter.